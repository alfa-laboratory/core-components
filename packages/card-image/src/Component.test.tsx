import React from 'react';
import { render } from '@testing-library/react';

import { CardImage, DEFAULT_BASE_URL, DEFAULT_WIDTH, ASPECT_RATIO } from './index';

const layers = 'BACKGROUND,CARD_NUMBER,CARD_HOLDER';
const cardId = 'ER';
describe('CardImage', () => {
    describe('Display tests', () => {
        it('should display correctly', () => {
            expect(render(<CardImage />)).toMatchSnapshot();
        });
        it('should display with cardId correctly', () => {
            expect(render(<CardImage cardId='ER' />)).toMatchSnapshot();
        });
        it('should display with cardId and layers', () => {
            const { container } = render(<CardImage cardId={cardId} layers={layers} />);
            const img = container.querySelector('img');
            expect(img?.src).toContain(cardId);
            expect(img?.src).toContain(`layers=${layers}`);
            expect(container).toMatchSnapshot();
        });
    });

    describe('Styles tests', () => {
        it('should set custom class', () => {
            const className = 'custom-class';
            const { container } = render(<CardImage className={className} />);

            expect(container.firstElementChild).toHaveClass(className);
        });

        it('should set `rounded` class', () => {
            const { container } = render(<CardImage rounded={true} />);

            expect(container.firstElementChild).toHaveClass('rounded');
        });
    });

    describe('Render tests', () => {
        test('should unmount without errors', () => {
            const { unmount } = render(<CardImage />);

            expect(unmount).not.toThrowError();
        });

        test('should have default baseUrl', () => {
            const { container } = render(<CardImage cardId={cardId} layers={layers} />);
            const img = container.querySelector('img');
            expect(img?.src).toContain(DEFAULT_BASE_URL);
        });

        test('should have baseUrl', () => {
            const baseUrl = 'some-url';
            const { container } = render(
                <CardImage cardId={cardId} layers={layers} baseUrl={baseUrl} />,
            );
            const img = container.querySelector('img');
            expect(img?.src).toContain(baseUrl);
        });

        test('should provide custom baseUrl', () => {
            const baseUrl = 'some-url';
            const { container } = render(
                <CardImage cardId={cardId} layers={layers} baseUrl={baseUrl} />,
            );
            const img = container.querySelector('img');
            expect(img?.src).toContain(baseUrl);
        });

        test('should have default width', () => {
            const { container } = render(<CardImage cardId={cardId} layers={layers} />);
            const img = container.querySelector('img');
            expect(img?.width).toBe(DEFAULT_WIDTH);
        });

        test('should correctly set height', () => {
            const width = 500;
            const height = width * ASPECT_RATIO;
            const { container } = render(
                <CardImage cardId={cardId} layers={layers} width={width} />,
            );
            const img = container.querySelector('img');
            expect(img?.height).toBe(height);
        });
    });
});
