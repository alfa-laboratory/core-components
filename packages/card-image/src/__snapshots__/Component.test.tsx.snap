// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardImage Display tests should display correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="cardImage rounded"
        style="width: 280px; height: 176.4px;"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="cardImage rounded"
      style="width: 280px; height: 176.4px;"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CardImage Display tests should display with cardId and layers 1`] = `
<div>
  <div
    class="cardImage rounded"
    style="width: 280px; height: 176.4px;"
  >
    <img
      class="image"
      height="176.4"
      role="presentation"
      src="https://online.alfabank.ru/cards-images/cards/ER/images?layers=BACKGROUND,CARD_NUMBER,CARD_HOLDER&width=280"
      srcset="https://online.alfabank.ru/cards-images/cards/ER/images?layers=BACKGROUND,CARD_NUMBER,CARD_HOLDER&width=560 2x"
      width="280"
    />
  </div>
</div>
`;

exports[`CardImage Display tests should display with cardId correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="cardImage rounded"
        style="width: 280px; height: 176.4px;"
      >
        <img
          class="image"
          height="176.4"
          role="presentation"
          src="https://online.alfabank.ru/cards-images/cards/ER/images?layers=BACKGROUND,CARD_NUMBER,CARD_HOLDER,PAY_PASS,CHIP,LOGO,PAYMENT_SYSTEM,RESERVED_1,RESERVED_2,VALID_DATE&width=280"
          srcset="https://online.alfabank.ru/cards-images/cards/ER/images?layers=BACKGROUND,CARD_NUMBER,CARD_HOLDER,PAY_PASS,CHIP,LOGO,PAYMENT_SYSTEM,RESERVED_1,RESERVED_2,VALID_DATE&width=560 2x"
          width="280"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="cardImage rounded"
      style="width: 280px; height: 176.4px;"
    >
      <img
        class="image"
        height="176.4"
        role="presentation"
        src="https://online.alfabank.ru/cards-images/cards/ER/images?layers=BACKGROUND,CARD_NUMBER,CARD_HOLDER,PAY_PASS,CHIP,LOGO,PAYMENT_SYSTEM,RESERVED_1,RESERVED_2,VALID_DATE&width=280"
        srcset="https://online.alfabank.ru/cards-images/cards/ER/images?layers=BACKGROUND,CARD_NUMBER,CARD_HOLDER,PAY_PASS,CHIP,LOGO,PAYMENT_SYSTEM,RESERVED_1,RESERVED_2,VALID_DATE&width=560 2x"
        width="280"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
